src/components/CountryPicker/CountrySelectorModal.js

```diff
@@ -34,12 +34,12 @@ const defaultProps = {
 };
 
 function filterOptions(searchValue, data) {
-    const trimmedSearchValue = searchValue.trim();
-    if (trimmedSearchValue.length === 0) {
+    const searchValueWithOnlyAlphabets = searchValue.toLowerCase().replaceAll(CONST.REGEX.COUNTRY_NAME_WITH_ONLY_ALPHABETS, '');
+    if (searchValueWithOnlyAlphabets.length === 0) {
         return [];
     }
 
-    return _.filter(data, (country) => country.text.toLowerCase().includes(searchValue.toLowerCase()));
+    return _.filter(data, (country) => country.searchValue.includes(searchValueWithOnlyAlphabets) || country.value.toLowerCase().includes(searchValueWithOnlyAlphabets));
```

Comment:
This can be reduced to
```suggestion
    return _.filter(data, (country) => `${country.value}${country.text}`.toLowerCase().includes(trimmedSearchValue));
```

Reply:
But even then, I think we can improve the search itself. How can we "prioritize" searching by contry code so that, for instance, United States would be the first item when you enter `US`?
<img width="380" alt="image" src="https://github.com/Expensify/App/assets/9059945/d1e26456-f3cb-401d-abc1-5ed232f13ae3">